src/ui/views/main/components/menus/user_card.tsx

```diff
@@ -11,21 +19,95 @@ export interface IUserCardProps {
 
 import "./list.scss";
 import "./user_card.scss";
+import { selectPrefsFlag } from "state/selectors/prefs";
+import { pickColorFromHash } from "lib/palette";
+import { Markdown } from "ui/components/common/markdown";
 export function UserCard(props: IUserCardProps) {
-    let nick = props.member?.nick;
+    const { LL } = useI18nContext();
 
     let cached_user = useRootSelector((state: ReadRootState) => {
         let active_party = state.chat.active_party;
         return selectCachedUser(state, props.user.id, active_party);
     });
 
+    let user_discriminator = props.user.discriminator.toString(16).toUpperCase().padStart(4, '0');
+    
+    let state = useStructuredSelector({
+        is_light_theme: selectPrefsFlag(UserPreferenceFlags.LightMode),
+        party: (state: ReadRootState) => {
+            let party_id = activeParty(state);
+            if(party_id) {
+                return state.party.parties[party_id];
+            }
+            return;
+        },
+    });
+
+    
+    let member_info = createMemo(() => {
+        let members = state.party?.members;
+        for(let member of Object.values(members)) {
+            if(member.user.id == props.user.id) {
```

Comment:
`party.members`  is a plain object where the key is the user id, so instead of this loop you can just do 

```js
let member_info = createMemo(() => state.party?.members[props.user.id]);
```